c_src/quicer_listener.c

```diff
@@ -282,77 +281,36 @@ listen2(ErlNifEnv *env, __unused_parm__ int argc, const ERL_NIF_TERM argv[])
 
   // Start build CredConfig from with listen opts
   QUIC_CREDENTIAL_CONFIG CredConfig;
-  CxPlatZeroMemory(&CredConfig, sizeof(CredConfig));
 
-  CredConfig.Flags = QUIC_CREDENTIAL_FLAG_NONE;
-
-  if (!parse_cert_options(env, options, &CredConfig))
-    {
-      return ERROR_TUPLE_2(ATOM_QUIC_TLS);
-    }
-
-  BOOLEAN is_verify = FALSE;
-  if (!parse_verify_options(env, options, &CredConfig, TRUE, &is_verify))
-    {
-      return ERROR_TUPLE_2(ATOM_VERIFY);
-    }
+#if defined(QUICER_USE_TRUSTED_STORE)
+  X509_STORE *trusted_store = NULL;
+  ret = eoptions_to_cred_config(env, options, &CredConfig, &trusted_store);
+#else
+  ret = eoptions_to_cred_config(env, options, &CredConfig, NULL);
+#endif // QUICER_USE_TRUSTED_STORE
 
-  if (!parse_cacertfile_option(env, options, &cacertfile))
+  if (!IS_SAME_TERM(ret, ATOM_OK))
     {
-      // TLS opt error not file content error
-      free(cacertfile);
-      cacertfile = NULL;
-      free_certificate(&CredConfig);
-      return ERROR_TUPLE_2(ATOM_CACERTFILE);
+      return ERROR_TUPLE_2(ret);
```

Comment:
Q: should we `free_certificate(&CredConfig)` here?

Reply:
`eoptions_to_cred_config` take cares of that.